app/scripts/metamask-controller.js

```diff
@@ -2958,6 +2948,20 @@ export default class MetamaskController extends EventEmitter {
     }
   }
 
+  /**
+   * Get a Uint8Array mnemonic from Buffer.
+   *
+   * @param {Buffer} mnemonic - The mnemonic phrase as a Buffer
+   * @returns {Uint8Array} The mnemonic phrase as a Uint8Array
+   */
+  _bufferToUint8Array(mnemonic) {
+    const indices = mnemonic
+      .toString()
+      .split(' ')
+      .map((word) => wordlist.indexOf(word));
+    return new Uint8Array(new Uint16Array(indices).buffer);
+  }
```

Comment:
This seems like almost the reverse of: https://github.com/MetaMask/metamask-extension/pull/19817#discussion_r1275435354

What do you think about:

``` suggestion
  /**
   * Encodes a BIP-39 mnemonic as the indices of words in the English BIP-39 wordlist.
   *
   * @param {Buffer} mnenonic - The BIP-39 mnemonic.
   * @returns {Buffer} The Unicode code points for the seed phrase formed from the words in the wordlist.
   */
  _convertMnemonicToWordlistIndices(mnemonic) {
    const indices = mnemonic
      .toString()
      .split(' ')
      .map((word) => wordlist.indexOf(word));
    return new Uint8Array(indices);
  }
```

It might be good to relocate this and the other function to some kind of utility file as they're related operations, but this seems fine for now.

Reply:
Makes sense! I just applied your suggestion. 
Do you think that makes sense to merge the two PRs and then move the two methods to another file in a later PR?

Reply:
Uhm, it looks like this doesn't work. I noticed that also in `@metamask/eth-hd-keyring` we convert indices to [Uint16Array before converting them finally to Uint8Array](https://github.com/MetaMask/eth-hd-keyring/blob/main/src/HDKeyring.ts#L94).

I'm not exactly sure why this is needed though